core/store/src/metadata.rs

```diff
@@ -56,14 +56,12 @@ fn set_db_metadata(
     let mut store_update = storage.get_store(temp).store_update();
     store_update.set(DBCol::DbVersion, VERSION_KEY, metadata.version.to_string().as_bytes());
     if metadata.version >= DB_VERSION_WITH_KIND {
-        #[allow(unused_mut)]
         let mut kind = metadata.kind;
-        #[cfg(feature = "cold_store")]
-        if matches!(temp, Temperature::Cold) || storage.has_cold() {
-            kind = Some(if matches!(temp, Temperature::Hot) { DbKind::Hot } else { DbKind::Cold });
+        if temp == Temperature::Cold {
+            kind = Some(DbKind::Cold);
         }
-        if let Some(kind) = kind.map(|kind| <&str>::from(kind).as_bytes()) {
-            store_update.set(DBCol::DbVersion, KIND_KEY, kind);
+        if let Some(kind) = kind {
+            store_update.set(DBCol::DbVersion, KIND_KEY, <&str>::from(kind).as_bytes());
```

Comment:
This is a real change, not just refactoring
- previously this code would change the archival kind to hot kind
- currently this code will keep the archival kind as archival
This is in line with the migration plan.